api/v1beta1/cloudstackaffinitygroup_conversion.go

```diff
@@ -30,3 +33,44 @@ func (dst *CloudStackAffinityGroup) ConvertFrom(srcRaw conversion.Hub) error { /
 	src := srcRaw.(*v1beta2.CloudStackAffinityGroup)
 	return Convert_v1beta2_CloudStackAffinityGroup_To_v1beta1_CloudStackAffinityGroup(src, dst, nil)
 }
+
+//nolint:golint,revive,stylecheck
+func Convert_v1beta1_CloudStackAffinityGroup_To_v1beta2_CloudStackAffinityGroup(in *CloudStackAffinityGroup, out *v1beta2.CloudStackAffinityGroup, s conv.Scope) error {
+	out.ObjectMeta = in.ObjectMeta
+
+	csCluster := &v1beta2.CloudStackCluster{}
+	clusterName := in.ObjectMeta.Labels["cluster.x-k8s.io/cluster-name"]
+	namespace := in.ObjectMeta.Namespace
+	key := client.ObjectKey{Name: clusterName, Namespace: namespace}
+	if err := v1beta2.K8sClient.Get(context.TODO(), key, csCluster); err != nil {
+		return err
+	}
+
+	out.Spec = v1beta2.CloudStackAffinityGroupSpec{
+		Type:              in.Spec.Type,
+		Name:              in.Spec.Name,
+		ID:                in.Spec.ID,
+		FailureDomainName: csCluster.Spec.FailureDomains[0].Name,
+	}
+
+	out.Status = v1beta2.CloudStackAffinityGroupStatus{
+		Ready: in.Status.Ready,
+	}
+	return nil
+}
+
+//nolint:golint,revive,stylecheck
+func Convert_v1beta2_CloudStackAffinityGroup_To_v1beta1_CloudStackAffinityGroup(in *v1beta2.CloudStackAffinityGroup, out *CloudStackAffinityGroup, s conv.Scope) error {
```

Comment:
It's interesting that we have these backward conversion methods, but they're only partially functional.

I wonder if we might want to go ahead with annotations as a way to make them fully functional?